compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala

```diff
@@ -62,8 +62,12 @@ class QuoteDriver extends Driver {
     output.getOrElse(throw new Exception("Could not extract " + expr))
   }
 
+  private[this] val myDefinitions = new ThreadLocal[Definitions] {
```

Comment:
I'm not convinced at all this is safe. `Definitions` contains state like the `ctx` field and the `_isInitialized` field, why is it correct to share this between every quote in the same JVM thread? Note that you may have more than one dotty compiler running on the same JVM thread.